File: src/kbot/src/Task_5/sensor_subscriber.py
#!/usr/bin/env python
import rospy
from kbot.msg import Sensor

def sensor_callback(sensor_message):
    rospy.loginfo("new data received: (%d, %s, %.2f ,%.2f)", 
        sensor_message.id,sensor_message.name,
        sensor_message.temperature,sensor_message.humidity)
    
rospy.init_node('sensor_subscriber_node', anonymous=True)

rospy.Subscriber("sensor_topic", Sensor, sensor_callback)

# spin() simply keeps python from exiting until this node is stopped
rospy.spin()

File: src/kbot/src/Task_5/Sensorsubscriber.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "kbot/Sensor.h"
void PublishMsg_callback(const kbot::Sensor& msg)
{
  std::cout<<"acquired Sensor id : "<<(int)msg.id<<"-- acquired Sensor name : "<<msg.name<<"-- acquired temperature : "<<(float)msg.temperature<<"-- acquired Sensor humdidity : "<<(float)msg.humidity<<std::endl;
}
int main(int argc, char **argv)
{
  ros::init(argc, argv, "Sensorsubmsg");
  ros::NodeHandle n;
  ros::Subscriber sensor_sub = n.subscribe("Sensorpubmsg", 1000, PublishMsg_callback);
  ros::spin();
  return 0;
}
